.gitignore

```diff
@@ -1 +1,10 @@
-_build
\ No newline at end of file
+_build
+
+# -------- CMake --------
+build/*
+
+# -------- IDE --------
+.vscode/*
+.vs/*
+out/*
+CMakeSettings.json

```

Comment:
Could you explain why this one is necessary?

Reply:
I do not know about out/CMakeSettings.json. I don't use them (as far as I know).

Reply:
This file also automatically created in MS VS 2019 in the root folder when you open solution from CMakeLists.txt file.
Also, this file already exist in example of Alexey Veprev.
Schema reference of this file described in https://docs.microsoft.com/en-us/cpp/build/cmakesettings-reference?view=msvc-170